### ISSUE TITLE
Make `--model` parameter of host application more flexible

### ISSUE THREAD

**Issue by hannobraun** [hannobraun/Fornjot#9]
Currently, if you run `fj-host -m my-model`, it tries to find a model in `models/my-model`. This is convenient for use in this repository, but it is less convenient for other use cases.

Here's how I could see that being fixed:
1. If no `-m` argument is passed and there is a Cargo package in the current directory, load that.
2. If no `-m` arugment is passed and there is no Cargo package in the current directory, fall back to current behavior.
3. If the parameter passed to `-m` is an existing path, attempt to load a model from there.

Labeling https://github.com/hannobraun/Fornjot/labels/good%20first%20issue, as this is a rather small change that doesn't require detailed knowledge of Fornjot. Some notes on the implementation:
- Arguments are specified in [`args.rs`](https://github.com/hannobraun/Fornjot/blob/main/src/args.rs).
- Models are loaded in [`model.rs`](https://github.com/hannobraun/Fornjot/blob/main/src/model.rs).

I think it would make sense to add the new logic in `model.rs`.

**Comment by hannobraun** [comment]
In case anyone wants to look into this:
- The argument is specified in [`args.rs`](https://github.com/hannobraun/fornjot/blob/06d90d846c38249466409a73e3a398f3d46de49f/src/args.rs#L6-L8).
- The argument is then used in [`model.rs`](https://github.com/hannobraun/fornjot/blob/06d90d846c38249466409a73e3a398f3d46de49f/src/model.rs#L18-L20).

I think it would make sense to add the logic I described here in `model.rs`.

**Comment by hannobraun** [comment]
I'm working on this.

**Comment by hannobraun** [comment]
I've looked into this, and it has turned out to be surprisingly difficult. The problem is, a model looks like any other Cargo package, so determining whether there is a model in the current directory isn't straight-forward. This could be solved by adding a special file to model directories, or special metadata to the `Cargo.toml` of models, but that's more complication for the user. I don't think that's worth it right now.

How models look is an area that is going to change a lot in the future, with the planned move to WASM (#71) and support for more modeling languages. Once that has settled down somewhat, I think it's worth revisiting this issue. For now, I've added it to the [feature wishlist](https://github.com/hannobraun/Fornjot/discussions/146).
